blog/2024-10-25-object-notifications/index.mdx

```diff
@@ -0,0 +1,133 @@
+---
+slug: how-we-built-object-notifications
+title: How we built object notifications in Tigris
+description: >
+  A deep dive into the implementation of object notifications in Tigris.
+
+image: ./trees.jpg
+keywords:
+  [object storage, blob storage, s3, ai, object notifications, event streaming]
+authors: [garren]
+tags: [object storage, s3, event streaming, changes, notifications]
+---
+
+![Autumn trees on a dusty road in Magoebaskloof, South Africa](./trees.jpg)
+
+<center>
+  <small>
+    <em>
+      Autumn trees on a dusty road in Magoebaskloof, South Africa. Photo by
+      Garren Smith, iPhone 13 Pro.
+    </em>
+  </small>
+</center>
+
+# How We Built Object Notifications in Tigris
+
+Tigris now supports object notifications! Object notifications are a way of
+receiving events for changes in a bucket. Think of it as your bucket's way of
+saying, "Hey, something happened! Come check it out!" These notifications can be
+helpful for keeping track of what's going on in your application.
+
+## Use Case: Automatic Image Processing
+
+A simple use case for object notifications is to automatically process images.
+Imagine you're building a photo-sharing app. Every time a user uploads a new
+picture, you want to automatically generate a thumbnail and maybe even run it
+through an AI to detect any inappropriate content. With object notifications,
+this becomes a breeze!
+
+Here's how it could work:
+
+1. User uploads an image to your Tigris bucket.
+2. Tigris sends a notification to your webhook.
+3. Your server receives the notification and springs into action.
+4. It downloads the new image, creates a thumbnail, and runs it through an AI
+   check.
+5. The processed image and its metadata are saved back to Tigris.
```

Comment:
I agree, this is a great use case to show.

Reply:
Not only for image, think of preprocessing of any raw data, would be applicable for AI workloads as well. Service receives an event, process the data and rewrite the data again with extra metadata/tags/etc.

Reply:
Write. Image upload moderation using AI is just such an obvious use case and cool to show.

Reply:
Other I can think about is when video is uploaded, converting it to frames. Kind of what Yoko's app is doing it but simplifying it.

Reply:
This might be the thing we do with hookdeck. Run it through hookdeck route it to AI and back to Tigris in another folder. It could possibly be a no-code process